main/src/library/File.flix

```diff
@@ -151,4 +200,97 @@ namespace File {
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path, ##java.nio.charset.Charset);
+            import java.nio.charset.Charset:forName(String);
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath, forName(opts.charSet));
+            Ok(readAll(reader))
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Err(getMessage(ex))
+        }
+
+    ///
+    /// Returns an iterator of the given file `f`
+    ///
+    pub def readLinesIter(f: String): Iterator[Result[String, String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path);
+            import java.io.BufferedReader.readLine();
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath);
+            let line = ref readLine(reader);
+
+            let done = () -> Object.isNull(deref line);
+            let next = () -> {
+                try {
+                    let l = deref line;
+                    line := readLine(reader);
+                    Ok(l)
+                } catch {
+                    case ex: ##java.io.IOException =>
+                    import java.lang.Throwable.getMessage();
+                    Err(getMessage(ex))
+                }
+            };
+            Iterator(done, next)
+        } catch {
+            case ex: ##java.io.IOException =>
+                import java.lang.Throwable.getMessage();
+                Iterator.singleton(Err(getMessage(ex)))
+        }
+
+    ///
+    /// Returns an iterator of the given file `f` with the options `opts`.
+    /// The options `opts` to apply consists of
+    /// `charSet` - the specific charset to be used to decode the bytes.
+    ///
+    pub def readLinesIterWith(opts: {charSet :: String}, f: String): Iterator[Result[String, String]] & Impure =
+        try {
+            import new java.io.File(String) as newFile;
+            import java.io.File.toPath();
+            import java.nio.file.Files:newBufferedReader(##java.nio.file.Path, ##java.nio.charset.Charset);
+            import java.nio.charset.Charset:forName(String);
+            import java.io.BufferedReader.readLine();
+
+            let javaFile = newFile(f);
+            let javaPath = toPath(javaFile);
+            let reader = newBufferedReader(javaPath, forName(opts.charSet));
+            let line = ref readLine(reader);
+
+            let done = () -> Object.isNull(deref line);
+            let next = () -> {
+                try {
+                    let l = deref line;
+                    line := readLine(reader);
+                    Ok(l)
+                } catch {
```

Comment:
@jaschdoc You see the programming pattern above? Do you think that we could support it inside iterator? Or maybe its a bad idea? (Something like unfolding a function as long as it returns non-null?)

Reply:
I like that idea. Let's experiment with it. Are you thinking something like `Iterator.ofNullable(next: Next[a], nullable: Ref[a])`. I'm not quite sure about the signature of the 2nd argument but maybe something like that could work.

Reply:
OK, how about you add it to your list and we can do it after I merge this?